drv/lpc55-update-server/src/main.rs

```diff
@@ -301,29 +504,43 @@ fn do_block_write(
         None => return Err(UpdateError::OutOfBounds),
     };
 
-    // Step one: erase the page.
+    // write_addr is a byte address; convert it back to a page number, but this
+    // time, an absolute page number in the flash device.
+    let page_num = write_addr / BYTES_PER_FLASH_PAGE as u32;
 
-    // write_addr is a byte address; page_num is a page index; but the actual
-    // erase machinery operates in terms of flash words. A flash word is 16
-    // bytes in length. So, we need to convert.
-    //
-    // Note that the range used here is _inclusive._
-    //
-    // We wind up needing this in u32 form a lot, so let's cast once and reuse.
-    const WORDSZ: u32 = BYTES_PER_FLASH_WORD as u32;
+    do_raw_page_write(flash, page_num, flash_page)
+}
+
+/// Performs an erase-write sequence to a single page within the raw flash
+/// device. This function is capable of writing outside of any image slot, which
+/// is important for doing CFPA updates. If you're writing to an image slot, use
+/// `do_block_write`.
+fn do_raw_page_write(
+    flash: &mut drv_lpc55_flash::Flash<'_>,
+    page_num: u32,
+    flash_page: &[u8; BLOCK_SIZE_BYTES],
```

Comment:
Since this function operates on pages, it would be more clear to use `&[u8; BYTES_PER_FLASH_PAGE]` here even though it's identical; this would also save us if someone tries to have `BLOCK_SIZE_BYTES` diverge from the page size.

Reply:
We're at the impedance boundary between my driver (which uses words from the datasheet) and the update code (which has invented the concept of blocks and applies them to systems that don't really have flash pages like the STM32H7). So, this is inconsistent, but it would be _real bad_ if they diverged.

I could go either way -- I copied this constant from `do_block_write`. Is it weirder to have the two differ or to use `BLOCK_SIZE_BYTES` here?

Reply:
I think it's weirder to use `BLOCK_SIZES_BYTES`.

I'd vote to keep `do_block_write` in terms of `BLOCK_SIZE_BYTES` and switch `do_raw_page_write` to use `BYTES_PER_FLASH_PAGE`; because the `flash_page` argument is passed straight through, we should get an error if they somehow diverged.

Reply:
Changing.